Develop

MakhnoGK opened the pull request:
Rewrite application features using Redux and Redux Saga

MakhnoGK committed:
Initial Redux setup with ducks approach

MakhnoGK committed:
Add redux saga

Basic search operation using redux saga approach

MakhnoGK committed:
Complete redux saga integration

MakhnoGK committed:
Update ImageItem.js

MakhnoGK committed:
Added redux-dev-tools extension

MakhnoGK committed:
Merge branch 'feature_redux' into develop

MakhnoGK committed:
Move tag manipulation functionality to SearchForm component

MakhnoGK committed:
Update search behoviour

Search requests are performed after tags added or removed now

MakhnoGK committed:
Minor code fixes

MakhnoGK committed:
Fix code style

MakhnoGK committed:
Fix wrong object destructuring in ImageItem component

mike-gusiev commented on src/components/App.js:
```diff
@@ -1,104 +1,70 @@
-  };
-
+const App = ({ imageData, loading }) => {
   return (
```
используй краткую форму es6 функций без return, там где это возможно

mike-gusiev commented on src/ducks/images.js:
```diff
@@ -0,0 +1,41 @@
+export const REQUEST_IMAGES = 'app/images/request';
```
труктура редакса немного неправильная. избавься от папки reducers. и создай папку src/redux/ducks. и туда в отдельные файлы помести все екшены,редюсеры/саги и тд одним файлом. все что относиться к одному duck - должно быть в одном файле:
https://github.com/erikras/ducks-modular-redux

а в src/redux/index.js должны быть все настройки стора редакса

MakhnoGK committed:
Fixes after review

- Changed redux structure
- Refactor component functions to use ES6 shorthand syntax without "return" keyword

MakhnoGK committed:
Fix useTags hook

- Changed logic of useTags hooks, so now it removes special characters from user input.

MakhnoGK committed:
Move API request to new module

MakhnoGK committed:
Merge branch 'master' into develop
